frontend/src/components/editor/alerts/connecting-alert.tsx

```diff
@@ -0,0 +1,21 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+
+import { useAtomValue } from "jotai";
+import { FloatingAlert } from "./floating-alert";
+import { isConnectingAtom } from "@/core/network/connection";
+import { Spinner } from "@/components/icons/spinner";
+
+const DELAY_MS = 1000; // 1 second
+
+export const ConnectingAlert: React.FC = () => {
+  const isConnecting = useAtomValue(isConnectingAtom);
+
+  return (
+    <FloatingAlert title="Connecting" show={isConnecting} delayMs={DELAY_MS}>
+      <div className="flex items-center gap-2">
+        <Spinner className="h-4 w-4" />
+        <p>Establishing a connection to a marimo runtime...</p>
+      </div>
+    </FloatingAlert>
+  );
```

Comment:
Would we want to present different states here, or just connecting?

Reply:
for now just connecting. what states did you have in mind? it might be nice to show not connected or maybe with a reconnect button (if/when we have that flow, currently we don't)